OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -0,0 +1,72 @@
+﻿using OWML.Common;
+using OWML.ModHelper.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OWML.ModHelper.Interaction
+{
+    public class ModInteraction : IModInteraction
+    {
+        private readonly List<ModBehaviour> _modList = new List<ModBehaviour>();
+        private readonly IModFinder _finder;
+        private readonly Dictionary<IModData, List<IModData>> dependantDict = new Dictionary<IModData, List<IModData>>();
+
+        public ModInteraction(List<ModBehaviour> list, IModFinder finder)
+        {
+            _modList = list;
+            _finder = finder;
+
+            foreach (var mod in _finder.GetMods())
+            {
+                List<IModData> temp = new List<IModData>();
+                foreach (var mod2 in _finder.GetMods())
+                {
+                    if (mod2.Manifest.Dependencies != null && mod2.Manifest.Dependencies.Contains(mod.Manifest.Name))
+                    {
+                        temp.Add(mod2);
+                    }
+                }
+                dependantDict.Add(mod, temp);
+            }
+        }
+
+        public IList<IModData> GetMods()
+        {
+            return _finder.GetMods();
+        }
+
+        public IList<IModData> GetDependants(string dependencyUniqueName)
+        {
+            return dependantDict.Where(x => x.Key.Manifest.UniqueName == dependencyUniqueName).FirstOrDefault().Value;
+        }
+
+        public bool ModExists(string uniqueName)
+        {
+            return _finder.GetMods().Where(m => m.Manifest.UniqueName == uniqueName).Count() != 0;
+        }
+
+        public T InvokeMethod<T>(string uniqueName, string methodName, params object[] parameters)
+        {
+            var mod = _modList.Where(m => m.ModHelper.Manifest.UniqueName == uniqueName).First();
+            return (T)mod.GetType().GetMethod(methodName).Invoke(mod, parameters);
+        }
+
+        public void InvokeMethod(string uniqueName, string methodName, params object[] parameters)
+        {
+            var mod = _modList.Where(m => m.ModHelper.Manifest.UniqueName == uniqueName).First();
+            mod.GetType().GetMethod(methodName).Invoke(mod, parameters);
```

Comment:
Should handle cases where .GetMethod() is null. This goes for many of these methods.